torch/_torch_docs.py

```diff
@@ -3152,7 +3152,24 @@ def merge_dicts(*dicts):
            r"""
 flatten(input, start_dim=0, end_dim=-1) -> Tensor
 
-Flattens a contiguous range of dims in a tensor.
+By default, flattens :attr:`input` into a one-dimensional tensor with the same number of elements.
+If :attr:`start_dim` or :attr:`end_dim` are passed, only dimensions included in the range are
+flattened.
+
+Flattening an :attr:`input` with shape :math:`(l*, f+, r*)` produces an output shape of
+:math:`(l*, \prod f+, r*)` where :math:`l*` is the sizes of the zero or more "left" dimensions,
+:math:`f+` is the sizes of the one or more dimensions between :attr:`start_dim` and :attr:`end_dim` that
+will be "flattened", and :math:`r*` is the sizes of the zero or more "right" dimensions in :attr:`input`.
```

Comment:
I would consider adding the next paragraph to this one. That way they'd both be under the start_dim != end_dim case, and there may be a more natural segue.

For example, a new last sentence for this paragraph could begin, 

"It is equivalent to calling torch.reshape(...), and like :func:torch.reshape returns a view when possible. See (torch.view reference) for details on when a view will be returned. "